orchagent/switchorch.cpp

```diff
@@ -136,19 +209,30 @@ void SwitchOrch::doAppSwitchTableTask(Consumer &consumer)
     {
         auto t = it->second;
         auto op = kfvOp(t);
+        bool retry = false;
 
         if (op == SET_COMMAND)
         {
-            bool retry = false;
-
             for (auto i : kfvFieldsValues(t))
             {
                 auto attribute = fvField(i);
 
                 if (switch_attribute_map.find(attribute) == switch_attribute_map.end())
                 {
-                    SWSS_LOG_ERROR("Unsupported switch attribute %s", attribute.c_str());
-                    break;
+                    // Check additionally 'switch_tunnel_attribute_map' for Switch Tunnel
+                    if (switch_tunnel_attribute_map.find(attribute) == switch_tunnel_attribute_map.end())
+                    {
+                        SWSS_LOG_ERROR("Unsupported switch attribute %s", attribute.c_str());
+                        break;
+                    }
+                    else
+                    {
+                        retry  = switchTunnelSetVxlanParams(i);
```

Comment:
* Probably you can make it simpler, without using additional ```retry``` variable.
* And please also add ```{}```.
* Looks like ```continue``` is not needed since it is end of the ```if``` and ```for``` code blocks and anyway it will go to the next loop iteration.
```
 if (switchTunnelSetVxlanParams(i))
 {
     break;
 }
```

Reply:
- Without `retry ` initialization this code after `for` loop will never be executed
```
if (retry == true)
{
    it++;
}
```
- without `continue ` iteration will go through all code below and will fail on line
 `attr.id = switch_attribute_map.at(attribute);`